index.js

```diff
@@ -52,4 +53,135 @@ const state = {
     }
   ],
   cart: []
-};
\ No newline at end of file
+};
+
+// SELECTORS
+const storeItemList = document.querySelector('.store--item-list')
+const cartItemList = document.querySelector('.cart--item-list')
+const totalNumber = document.querySelector('.total-number')
+
+
+const clearStorePage = () => {
+  storeItemList.innerHTML = ""
+}
+
+const clearCart = () => {
+  cartItemList.innerHTML = ""
+}
+
+const renderAllTheItemsInTheStore = () => {
+  clearStorePage()
+  state.items.forEach(currentItem => {
+    const newItem = document.createElement('li')
+    storeItemList.append(newItem)
+
+    const iconDiv = document.createElement('div')
+    iconDiv.setAttribute('class', 'store--item-icon')
+    storeItemList.append(iconDiv)
+
+    const img = document.createElement('img')
+    img.setAttribute('src', `assets/icons/${currentItem.id}.svg`)
+    img.setAttribute('alt', currentItem.name)
+    iconDiv.append(img)
+
+    const addToCartButton = document.createElement('button')
+    addToCartButton.innerText = 'Add to cart'
+    newItem.append(addToCartButton)
+
+    addToCartButton.addEventListener('click', () => {
+      addItemToCart(currentItem)
+      renderAllTheItemsInTheCart()
+      sumOfCart()
+    })
+  })
+}
+
+let alreadyInCart = false
+const checkIfItemIsInCart = (item) => {
+    for (i = 0; i < state.cart.length; i++) {
+      if (state.cart[i].name === item.name) {
+        alreadyInCart = true
+        state.cart[i].quantity += 1
+        break
+      } else {
+        alreadyInCart = false
+      }
+    }
+}
+
+const addItemToCart = (product) => {
+  const newItemObj = {
+    name: product.name,
+    price: product.price,
+    img: `assets/icons/${product.id}.svg`,
+    quantity: 1
+  }
+  checkIfItemIsInCart(product)
+  if (!alreadyInCart) {
+    state.cart.push(newItemObj)
+  }
+}
+
+const renderAllTheItemsInTheCart = () => {
+  clearCart()
+  state.cart.forEach(cartItem => {
+    const newCartItem = document.createElement('li')
+    cartItemList.append(newCartItem)
+
+    const img = document.createElement('img')
+    img.setAttribute('class', 'cart--item-icon')
+    img.setAttribute('src', cartItem.img)
+    img.setAttribute('alt', cartItem.name)
+    newCartItem.append(img)
+
+    const name = document.createElement('p')
```

Comment:
I like how you organise each element, with each one having all of it's actions blocked together (i.e. making the const, setting the attribute, and appending all in a block)